pkg/grpc/podstore/podstore.go

```diff
@@ -246,7 +246,7 @@ func (s store) MarkPodFailed(_ context.Context, req *podstore_protos.MarkPodFail
 	// we don't really need the CAS properties of MutateStatus but there
 	// should be only one system trying to write the status record so it
 	// doesn't hurt.
-	err = s.podStatusStore.MutateStatus(podUniqueKey, mutator)
+	err = s.podStatusStore.MutateStatus(ctx, podUniqueKey, mutator)
```

Comment:
TODO: This is still not fixed. I'm confused why the test for this passes.

See: https://github.com/square/p2/blob/master/pkg/grpc/podstore/podstore_test.go#L426-L460